src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -1341,26 +1333,36 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
 
                 if (oldIndex != -1)
                 {
-                    windowEditorObjectSelectionSelectObject(6, oldObject._header);
+                    ObjectManager::ObjectSelectionMeta meta{};
+                    std::copy(std::begin(_112C1C5), std::end(_112C1C5), std::begin(meta.numSelectedObjects));
+                    meta.numImages = _112C209;
+                    ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultDeselect, *oldObject._header, selectionFlags, meta);
+                    std::copy(std::begin(meta.numSelectedObjects), std::end(meta.numSelectedObjects), std::begin(_112C1C5));
+                    _112C209 = meta.numImages;
                 }
             }
         }
 
-        auto bx = 0;
+        auto mode = ObjectManager::SelectObjectModes::defaultDeselect;
 
         if ((selectionFlags[index] & ObjectManager::SelectedObjectsFlags::selected) == ObjectManager::SelectedObjectsFlags::none)
         {
-            bx |= (1 << 0); // Selection mode
+            mode = ObjectManager::SelectObjectModes::defaultSelect;
         }
 
-        bx |= 6;
+        ObjectManager::ObjectSelectionMeta meta{};
+        std::copy(std::begin(_112C1C5), std::end(_112C1C5), std::begin(meta.numSelectedObjects));
```

Comment:
Do you have a name for `_112C1C5`? I can make a PR to move it to a getter+setter

Reply:
As far as i can work out it's just a temp global for the options window. I modify a few things related to it in the next few PRs so can address the name issue at the end of those